lib/addons/vpc-cni/index.ts

```diff
@@ -1,15 +1,248 @@
+  */
+  disableIntrospection?: boolean;
+  /**
+  * `DISABLE_METRICS` Environment Variable. Type: Boolean.
+  * Specifies whether the prometheus metrics endpoint is disabled or not for 
+  * ipamd. By default metrics are published on :61678/metrics.
+  */
+  disableMetrics?: boolean;
+  /**
+  * `DISABLE_NETWORK_RESOURCE_PROVISIONING` Environment Variable. Type: Boolean.
+  * Setting DISABLE_NETWORK_RESOURCE_PROVISIONING to true will make IPAMD 
+  * depend only on IMDS to get attached ENIs and IPs/prefixes.
+  */
+  disablenetworkResourceProvisioning?: boolean;
+  /**
+  * `ENABLE_POD_ENI` Environment Variable. Type: Boolean.
+  * Setting ENABLE_POD_ENI to true will allow IPAMD to add the 
+  * vpc.amazonaws.com/has-trunk-attached label to the node if the instance 
+  * has the capacity to attach an additional ENI.
+  */
+  enablePodEni?: boolean;
+  /**
+  * `ENABLE_PREFIX_DELEGATION` Environment Variable. Type: Boolean.
+  * To enable prefix delegation on nitro instances. Setting 
+  * ENABLE_PREFIX_DELEGATION to true will start allocating a prefix (/28 for IPv4 and /80 for IPv6) instead of a secondary IP in the ENIs subnet. 
+  */
+  enablePrefixDelegation?: boolean;
+  /**
+  * `WARM_ENI_TARGET` Environment Variable. Format integer.
+  * Specifies the number of free elastic network interfaces (and all of their 
+  * available IP addresses) that the ipamd daemon should attempt to keep 
+  * available for pod assignment on the node. 
+  */
+  warmEniTarget?: number;
+  /**
+  * `WARM_PREFIX_TARGET` Environment Variable. Format integer.
+  * Specifies the number of free IPv4(/28) prefixes that the ipamd daemon 
+  * should attempt to keep available for pod assignment on the node.
+  */
+  warmPrefixTarget?: number;
+  /**
+   * Secondary Subnet IDs for creating `ENIConfig`
+   */
+  customNetworkingConfig?: CustomNetworkingConfig;
+}
+
+
+export interface CustomNetworkingConfig {
+  /**
+   * Explicitly select individual subnets
+   * @default - Use all subnets in a selected group (all private subnets by default)
+   */
+  readonly subnets?: ISubnet[];
+}
+
+const defaultProps: CoreAddOnProps = {
+  addOnName: 'vpc-cni',
+  version: 'v1.12.1-eksbuild.2',
+  saName: 'vpc-cni',
+  namespace: 'kube-system',
+  controlPlaneAddOn: true,
+  configurationValues: {}
+};
+
+/**
+ * Implementation of VpcCni EKS add-on with Advanced Configurations.
  */
 export class VpcCniAddOn extends CoreAddOn {
+    
+  readonly vpcCniAddOnProps: VpcCniAddOnProps;
+  readonly id? : string;
+
+  constructor(props?: VpcCniAddOnProps) {
+    super({...defaultProps, ...props});
+    this.vpcCniAddOnProps = { ...defaultProps, ...props, };
+    (this.coreAddOnProps.configurationValues as any) = populateVpcCniConfigurationValues(props);
+  }
+
+  deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+      const cluster = clusterInfo.cluster;
+      let clusterSecurityGroupId = cluster.clusterSecurityGroupId;
+      let doc: string;
```

Comment:
why do you need this separate declaration?

Reply:
Done